src/include/miopen/hipoc_kernel.hpp

```diff
@@ -47,29 +48,20 @@ inline HipEventPtr make_hip_event()
 
 #if 1 // Keep around other storage techinques -- @pfultz2 27.03.2017
 
-#if 1 // Keep around other storage techinques -- @pfultz2 27.03.2017
 template <class T, class U>
 struct KernelArgsPair
 {
-    static const int alignment    = sizeof(U);
-    static const int padding      = (alignment - sizeof(T) % alignment) % alignment;
-    static const int second_index = sizeof(T) + padding;
+    constexpr static const auto alignU       = alignof(U);
+    constexpr static const auto padding      = (alignU - (sizeof(T) % alignU)) % alignU;
+    constexpr static const auto second_index = sizeof(T) + padding;
     KernelArgsPair(T x, U y)
     {
         new(buffer) T(x); // NOLINT (clang-analyzer-cplusplus.PlacementNew)
         new(buffer + second_index) U(y);
     }
+
     alignas(U) char buffer[second_index + sizeof(U)] = {};
 };
-#else
```

Comment:
Ditto

Reply:
duplicate of https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2334#discussion_r1339199244
[resolved]